lib/stackprof/report.rb

```diff
@@ -80,65 +83,36 @@ def print_stackcollapse
       end
     end
 
-    def print_flamegraph(f=STDOUT, skip_common=true)
```

Comment:
We need to maintain this API.  We're using flamegraphs in production by serving a static asset (the flamegraph viewer) and it makes a request to and endpoint that serves up the flamegraph JSON for a particular page.  We use this method to print the flamegraph JSON to the socket

Reply:
Can you elaborate on the workflow you use? This function as it currently exists doesn't print JSON directly, but rather prints a JavaScript function invocation whose only argument happens to be valid JSON.

Do you take the output of this and strip the `flamegraph(` at the beginning and the trailing `)` at the end then serve it as true JSON?

Or, when you say 
> it makes a request to and endpoint that serves up the flamegraph JSON for a particular page
do you mean that you set up the flamegraph viewer to include a `<script>` which references the exact output of `print_flamegraph`?

Both before and after this PR, the data and the viewer are separated, and before and after this PR the data is valid JavaScript but not valid JSON without modification.

So I *think* the request here is that "the step for extracting the JavaScript file and the step for opening the browser need to be separate steps so that consumers of stackprof as a library can serve the JavaScript file manually".

If that's the request, I'm going to open up a different code pathway for this.

Speedscope has multiple ways of loading data.
1. Dropping local files in
2. Browsing for local files
3. Specifying files via (possibly CORS) XHR URLs, which would *must not* contain the JavaScript function invocation wrapper (documented here: https://github.com/jlfwong/speedscope#importing-via-url)
4. Specifying files via `file://` protocol script URLs which *must* contain the Javascript function invocation wrapper, which is what this PR does since you can't make XHR requests to `file://` protocol URLs.

So if I understand you correctly, here's a proposed remediation:
1. Add a method called `get_speedscope_json` which returns a JSON string (not a JavaScript string) which could then be served over HTTP
2. Use that method in `view_flamegraph_in_browser` to retain the behavior in the PR as it stands

Then to upgrade your integration with stackprof, it would require changing the call-site to use `get_speedscope_json` instead of `print_flamegraph_json`, and also change the static assets that are being served right now to serve speedscope rather than serving the existing flamegraph viewer.

How does that path forward sound to you?

Reply:
Yes, this sounds perfect!

Reply:
This should be complete now